### PR TITLE
Refactoring, move os and os.exec wrapper into separate packages

### PR DESCRIPTION
**Pull request by flashvoid**
This change moves os and os.exec wrappers, implemented inside agent, out of agent and as separate packages. Allowing other components to reuse them.

### CODE REVIEW

**Comment by flashvoid** [comment]
Lint is complaining on 'Unkeyed composite literals' in 'agent/firewall_test.go'. It is not introduced problem and need to be fixed as separate task.

This change is about packaging only.

File: agent/helpers.go [hunk 1]
```diff
@@ -28,14 +28,17 @@ import (
...
 )
 
 // NewAgentHelper returns Helper with initialized default implementations
 // for all interfaces.
 func NewAgentHelper(agent *Agent) Helper {
 	helper := new(Helper)
-	helper.Executor = new(DefaultExecutor)
-	helper.OS = new(DefaultOS)
+	helper.Executor = new(utilexec.DefaultExecutor)
+	helper.OS = new(utilos.DefaultOS)
```
Comment (jbrendel) [inline agent/helpers.go]: OS? As in "operating system"? Why is this information needed?

Comment (flashvoid) [inline agent/helpers.go]: OS as in "operating system" but also as a name of module we are wrapping - "os" from standard library.
Now, while this component was a part of agent we needed a suffix after "Default" which is now obsolete - and should be "utilsos.Default" and "utilexec.Default". 

This current change, however, only about packaging. I deliberately skipped naming issue in order to keep this commit small. Naming need to be addressed but not in this change.

**Comment by jbrendel** [comment]
Don't see anything wrong with it.

I wish github's diff could be a bit smarter about refactoring where code is just moved around. I see tons of 'new' code, but it's really just the old code in a different place. Those make difficult reviews.

**Comment by pritesh** [comment]
:+1:

**Pull request merged**